backend/ml_model/use_cases/model.py

```diff
@@ -16,55 +16,93 @@
 from ml_model.repository.file_reader import FileReader
 from ml_model.entities.datapoint_entity import DataPoint
 from ml_model.repository.model_saver import save_model
-from ml_model.repository.fairness import FairnessEvaluator
 from ml_model.repository.data_preprocessing import DataProcessor
 
 current_dir = os.path.dirname(os.path.abspath(__file__))
 csv_file_path = os.path.join(current_dir, "../../../database/output.csv")
 
+from sklearn.model_selection import train_test_split, KFold
+from sklearn.exceptions import NotFittedError
+
+def safe_train_test_split(inputs, target, test_size=0.2, random_state=48):
```

Comment:
Can you explain these changes, with the 2 functions that you added?

Reply:
Oh yeah my bad forgot to mention that. It's making sure the split is never bigger than the sample size so the model doesn't crash.

Reply:
That's a good addition as the model was crashing is some cases where the data we had in output.csv for less.